src/freenet/node/useralerts/N2NTMUserAlert.java

```diff
@@ -22,38 +22,50 @@
 	private final long composedTime;
 	private final long sentTime;
 	private final long receivedTime;
+	private final long msgid;
 	private String sourceNodeName;
 	private String sourcePeer;
 
-	public N2NTMUserAlert(DarknetPeerNode sourcePeerNode,
-			String message, int fileNumber, long composedTime, long sentTime,
-			long receivedTime) {
+	public N2NTMUserAlert(DarknetPeerNode sourcePeerNode, String message, int fileNumber, long composedTime,
+			long sentTime, long receivedTime, long msgid) {
 		super(true, null, null, null, null, UserAlert.MINOR, true, null, true, null);
 		this.messageText = message;
 		this.fileNumber = fileNumber;
 		this.composedTime = composedTime;
 		this.sentTime = sentTime;
 		this.receivedTime = receivedTime;
-		peerRef = sourcePeerNode.getWeakRef();
-		sourceNodeName = sourcePeerNode.getName();
-		sourcePeer = sourcePeerNode.getPeer().toString();
+		this.peerRef = sourcePeerNode.getWeakRef();
+		this.sourceNodeName = sourcePeerNode.getName();
+		this.sourcePeer = sourcePeerNode.getPeer().toString();
+		this.msgid = msgid;
+	}
+
+	public N2NTMUserAlert(DarknetPeerNode sourcePeerNode, String message, int fileNumber, long composedTime,
+			long sentTime, long receivedTime) {
+		super(true, null, null, null, null, UserAlert.MINOR, true, null, true, null);
+		this.messageText = message;
+		this.fileNumber = fileNumber;
+		this.composedTime = composedTime;
+		this.sentTime = sentTime;
+		this.receivedTime = receivedTime;
+		this.peerRef = sourcePeerNode.getWeakRef();
+		this.sourceNodeName = sourcePeerNode.getName();
+		this.sourcePeer = sourcePeerNode.getPeer().toString();
+		this.msgid = Long.MAX_VALUE;
```

Comment:
Is there a specific reason for choosing `Long.MAX_VALUE` as magic value for a missing msgid? I would prefer using `0` or `-1` but it’s not a deal breaker. :)

Reply:
nope, using Math.abs and -1 would have been nicer. I changed that.